src/ray/gcs/gcs_server/gcs_task_manager.cc

```diff
@@ -407,18 +417,47 @@ void GcsTaskManager::HandleGetTaskEvents(rpc::GetTaskEventsRequest request,
       return false;
     }
 
-    if (filters.has_actor_id() && task_event.task_info().has_actor_id() &&
-        ActorID::FromBinary(task_event.task_info().actor_id()) !=
-            ActorID::FromBinary(filters.actor_id())) {
+    if (filters.task_filters_size() > 0) {
+      for (const auto &task_filter_obj : filters.task_filters()) {
+        if (task_filter_obj.predicate() == rpc::FilterPredicate::NOT_EQUAL &&
+            TaskID::FromBinary(task_event.task_id()) ==
+                TaskID::FromBinary(task_filter_obj.task_id())) {
+          return false;
+        }
+      }
+    }
+
+    if (filters.has_job_filter() &&
```

Comment:
I might miss some historical context. Here is seem that the original logic assumes only either task id or job id will be specified. So I updated here to make sure that both filters can be applied